### PR TITLE
Add validate to useEffect dependency array in useField

### PR DESCRIPTION
**Pull request by sventschui**

### CODE REVIEW

**Comment by JoviDeCroock** [comment]
Do you think that changing the `validate` function should trigger a validation?

I.E.: We have mounted and change the validate function (inside of a `useCallback`) because a dependent value has changed?

File: src/hooks/useField.ts [hunk 1]
```diff
@@ -31,14 +31,28 @@ export default function useField<T = any>(
...
+      // if the validation function changed, we will re-validate the field
+      const newError = validate(get(ctx.values, fieldId));
+      const oldError = get(ctx.errors, fieldId);
+
+      // we only invoke the setFieldError function when the error really changed, otherwise we might
+      // end up in an infinite loop
+      if (newError != oldError) {
+        ctx.setFieldError(fieldId, newError);
+      }
+    } else if (get(ctx.errors, fieldId)) {
```
Comment (sventschui) [inline src/hooks/useField.ts:45]: @JoviDeCroock not sure whether we should check for `!= undefined` here. Is `false` and/or `''`/`0` treated as an error?

Comment (JoviDeCroock) [reply src/hooks/useField.ts:45]: No this is good

File: src/hooks/useField.ts [hunk 2]
```diff
@@ -31,7 +31,21 @@ export default function useField<T = any>(
   useEffect(() => {
     const tuple: ValidationTuple = [fieldId, validate as (v: T) => string];
 
-    if (validate) ctx._fieldValidators.push(tuple);
+    if (validate) {
+      ctx._fieldValidators.push(tuple);
+      // if the validation function changed, we will re-validate the field
+      const newError = validate(get(ctx.values, fieldId));
```
Comment (JoviDeCroock) [inline src/hooks/useField.ts:37]: We might want to put this behind an isMounted flag

**Pull request merged**